src/Blef.GameLogic/PokerHands/TwoPairs.cs

```diff
@@ -1,27 +1,44 @@
-﻿using System.Linq;
+﻿using System;
+using System.Linq;
 
 namespace Blef.GameLogic.PokerHands
 {
     public class TwoPairs : PokerHand
     {
-        private readonly Rank first;
-        private readonly Rank second;
+        private readonly Rank firstRank;
+        private readonly Rank secondRank;
+        private const  int StrongerRankCoefficient = 10;
+        private const int BasicValue = 100;
 
-        public TwoPairs(Rank first, Rank second)
+        public TwoPairs(Rank firstRank, Rank secondRank)
         {
-            this.first = first;
-            this.second = second;
+            if (firstRank == secondRank)
+            {
+                throw new ArgumentException("Both ranks cannot be the same", nameof(secondRank));
+            }
+            this.firstRank = firstRank;
+            this.secondRank = secondRank;
         }
 
         public override bool IsOnTable(Table table)
         {
-            var firstRankCount = table.GetAllCards().Count(x => x.Rank == first);
-            var secondRankCount = table.GetAllCards().Count(x => x.Rank == second);
+            var firstRankCount = table.GetAllCards().Count(x => x.Rank == firstRank);
+            var secondRankCount = table.GetAllCards().Count(x => x.Rank == secondRank);
             return firstRankCount >= 2 && secondRankCount >= 2;
         }
 
-        protected override int Value => 100 + GetRankValue(first) + GetRankValue(second);
-
+        protected override long Value
+        {
+            get
+            {
+                var sortedRankValues = new[] {GetRankValue(firstRank), GetRankValue(secondRank)}.OrderBy(x => x).ToArray();
```

Comment:
It is not readable. For me it's better to enforce that first rank is greater in ctor (even maybe rename it to "greaterPair" or sth) and not have a need to do an ordering of two values here.
Even when playing real Poker you specify the greater rank first.